parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -497,7 +523,30 @@ pub mod pallet {
 			let beacon_block_root: H256 =
 				merkleization::hash_tree_root_beacon_header(header.clone())
 					.map_err(|_| Error::<T>::HeaderHashTreeRootFailed)?
-					.into(); // TODO check denylist headers
+					.into();
+
+			let finalized_block_root_hash =
```

Comment:
IMHO `finalized_block_root_hash` a little confusing here, would it better to rename as `block_roots_hash` or  some other to repensent not the root of the beacon block but the root of the blockRoots in the beacon state?